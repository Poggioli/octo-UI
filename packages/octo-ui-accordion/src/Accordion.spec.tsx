import { render } from "@testing-library/react";
import {
  Accordion,
  AccordionContent,
  AccordionHeader,
  AccordionItem,
} from "./Accordion";

describe("<Accordion />", () => {
  it(`GIVEN a <Accordion />
      WHEN render is called
      THEN should render a <Accordion />`, () => {
    const rendered = render(
      <Accordion type="multiple" value={["item-1", "item-2"]}>
        <AccordionItem disabled value="item-1">
          <AccordionHeader>Item 1 Header</AccordionHeader>
          <AccordionContent>Item 1 Content</AccordionContent>
        </AccordionItem>
        <AccordionItem value="item-2">
          <AccordionHeader>Item 2 Header</AccordionHeader>
          <AccordionContent>Item 2 Content</AccordionContent>
        </AccordionItem>
        <AccordionItem value="item-3">
          <AccordionHeader>Item 3 Header</AccordionHeader>
          <AccordionContent>Item 3 Content</AccordionContent>
        </AccordionItem>
        <AccordionItem disabled value="item-4">
          <AccordionHeader>Item 4 Header</AccordionHeader>
          <AccordionContent>Item 4 Content</AccordionContent>
        </AccordionItem>
      </Accordion>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Accordion />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Accordion.toString()).toBe(".octo-UI-c-dpBPGy");
  });

  it(`GIVEN a <AccordionItem />
      WHEN toString is called
      THEN return a string`, () => {
    expect(AccordionItem.toString()).toBe(".octo-UI-c-dzOJLW");
  });

  it(`GIVEN a <AccordionHeader />
      WHEN toString is called
      THEN return a string`, () => {
    expect(AccordionHeader.toString()).toBe(".octo-UI-c-hxvpuA");
  });

  it(`GIVEN a <AccordionContent />
      WHEN toString is called
      THEN return a string`, () => {
    expect(AccordionContent.toString()).toBe(".octo-UI-c-iGCJiJ");
  });
});
