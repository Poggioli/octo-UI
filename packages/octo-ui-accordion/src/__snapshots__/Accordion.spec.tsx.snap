// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> GIVEN a <Accordion />
      WHEN render is called
      THEN should render a <Accordion /> 1`] = `
<div>
  <div
    class="octo-UI-c-dpBPGy octo-UI-c-dpBPGy-crwvoE-shaded-true octo-UI-c-dpBPGy-bKVAaU-bordered-true"
    data-orientation="vertical"
  >
    <div
      class="octo-UI-c-dzOJLW"
      data-disabled=""
      data-orientation="vertical"
      data-state="open"
    >
      <h3
        class="octo-UI-c-hxvpuA"
        data-disabled=""
        data-orientation="vertical"
        data-state="open"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="true"
          class="octo-UI-c-fxJQfU"
          data-disabled=""
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="open"
          disabled=""
          id="radix-:r0:"
          type="button"
        >
          Item 1 Header
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="octo-UI-c-iGCJiJ"
        data-disabled=""
        data-orientation="vertical"
        data-state="open"
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
      >
        <div
          class="octo-UI-c-gqwkJN octo-UI-c-PJLV octo-UI-c-gqwkJN-djZMnX-flexDirection-initial octo-UI-c-gqwkJN-eJbcS-flexWrap-initial octo-UI-c-gqwkJN-fVplCG-justifyContent-initial octo-UI-c-gqwkJN-ebbZJG-alignItems-initial octo-UI-c-gqwkJN-dYGMXV-alignContent-initial octo-UI-c-gqwkJN-bexZoo-alignSelf-initial octo-UI-c-PJLV-iNyvjn-padded-true"
        >
          Item 1 Content
        </div>
      </div>
    </div>
    <div
      class="octo-UI-c-dzOJLW"
      data-orientation="vertical"
      data-state="open"
    >
      <h3
        class="octo-UI-c-hxvpuA"
        data-orientation="vertical"
        data-state="open"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="true"
          class="octo-UI-c-fxJQfU"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="open"
          id="radix-:r2:"
          type="button"
        >
          Item 2 Header
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="octo-UI-c-iGCJiJ"
        data-orientation="vertical"
        data-state="open"
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
      >
        <div
          class="octo-UI-c-gqwkJN octo-UI-c-PJLV octo-UI-c-gqwkJN-djZMnX-flexDirection-initial octo-UI-c-gqwkJN-eJbcS-flexWrap-initial octo-UI-c-gqwkJN-fVplCG-justifyContent-initial octo-UI-c-gqwkJN-ebbZJG-alignItems-initial octo-UI-c-gqwkJN-dYGMXV-alignContent-initial octo-UI-c-gqwkJN-bexZoo-alignSelf-initial octo-UI-c-PJLV-iNyvjn-padded-true"
        >
          Item 2 Content
        </div>
      </div>
    </div>
    <div
      class="octo-UI-c-dzOJLW"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="octo-UI-c-hxvpuA"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="octo-UI-c-fxJQfU"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          Item 3 Header
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="octo-UI-c-iGCJiJ"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="octo-UI-c-dzOJLW"
      data-disabled=""
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="octo-UI-c-hxvpuA"
        data-disabled=""
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r7:"
          aria-expanded="false"
          class="octo-UI-c-fxJQfU"
          data-disabled=""
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          disabled=""
          id="radix-:r6:"
          type="button"
        >
          Item 4 Header
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r6:"
        class="octo-UI-c-iGCJiJ"
        data-disabled=""
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r7:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
